### PR TITLE
Migrate to GitHub Actions and change install script to environment files

### LINKED ISSUE #4214: Migrate CI to GitHub Actions
Due to changes in the Travis CI billing, the Dask org is migrating CI to GitHub Actions.

This repo contains a `.travis.yml` file which needs to be replaced with an equivalent `.github/workflows/ci.yml` file.

See dask/community#107 for more details.

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Closes #4214 

This PR migrates CI to GitHub Actions.

Changes:
- All python versions are now tested on macos
- Environments are now built from environment files instead of many conda install commands
- Coverage is run on Python 3.8 Ubuntu only

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      matrix:
+        os: ["windows-latest", "ubuntu-latest", "macos-latest"]
+        python-version: ["3.6", "3.7", "3.8"]
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Setup Conda Environment
+        uses: goanpeca/setup-miniconda@v1
```
Comment (gforsyth) [inline .github/workflows/ci.yaml:19]:
```suggestion
        uses: conda-incubator/setup-miniconda@v2
```

**Review by gforsyth — COMMENTED** [review]
Looks good @jacobtomlinson -- do you want to merge from master?  I _think_ those windows tests should be passing now (although I also think they should've passed after you pinned `pytest-asyncio`, so...)

**Comment by jacobtomlinson** [comment]
Thanks @gforsyth.

I think we may still be blocked here on some SSH tests which are failing. I think GitHub Actions causes some problem with running our mock SSH server.

File: setup.cfg [hunk 2]
```diff
@@ -42,5 +42,5 @@ filterwarnings =
 minversion = 3.2
 markers =
     slow: marks tests as slow (deselect with '-m "not slow"')
-    avoid_travis: marks tests as flaky on TravisCI.
+    avoid_ci: marks tests as flaky on TravisCI.
```
Comment (jakirkham) [inline setup.cfg:45]:
```suggestion
    avoid_ci: marks tests as flaky on CI.
```

**Comment by jakirkham** [comment]
Pushed a commit to fix the merge conflict. Hope that is ok 🙂

**Comment by jacobtomlinson** [comment]
Thanks @jakirkham !

**Comment by jakirkham** [comment]
Ofc! Do you have a sense of the current issues we are running into here? I looked at CI, but nothing obvious stuck out at me.

File: continuous_integration/environment-3.7.yaml [hunk 3]
```diff
@@ -3,32 +3,43 @@ channels:
   - conda-forge
   - defaults
 dependencies:
-  - zstandard
-  - bokeh!=2.0.0
+  - python=3.7
+  - pip
+  - asyncssh
+  - bokeh
```
Comment (jakirkham) [inline continuous_integration/environment-3.7.yaml:9]: It looks like the `!=2.0.0` was left off here. Not sure if that was intended. There are a few other environment files missing this detail as well.

Comment (jacobtomlinson) [reply continuous_integration/environment-3.7.yaml:9]: Thanks John. Not sure why this was dropped.

**Comment by crusaderky** [comment]
Superseded by #4504; please close

**Pull request closed without merging**